### ISSUE TITLE
Unexpected results from curly: multi

### ISSUE THREAD

**Issue by RyanZim** [eslint/eslint#6919]
**What version of ESLint are you using?**

3.3.1

**What parser (default, Babel-ESLint, etc.) are you using?**

Default

**Please show your full configuration:**

```
module.exports = {
    "env": {
        "node": true
    },
    "extends": "eslint:recommended",
    "rules": {
        "indent": [
            "error",
            2
        ],
        "linebreak-style": [
            "error",
            "unix"
        ],
        "quotes": [
            "error",
            "single"
        ],
        "semi": [
            "error",
            "always"
        ],
       "curly": [
          "error",
          "multi"
       ]
    }
};
```

**What did you do? Please include the actual source code causing the issue.**

```
/* global foo, cb */
if (foo) {
  try {
    foo();
  } catch (err) {
    cb(err);
  }
}
```

**What did you expect to happen?**

I would expect the above code to be valid.

**What actually happened? Please include the actual, raw output from ESLint.**

```
$ eslint index.js

/home/ryan/t-eslint/index.js
  2:1  error  Unnecessary { after 'if' condition  curly

✖ 1 problem (1 error, 0 warnings)
```

---

If this is the expected behavior, what is the correct way to format the code?

**Comment by platinumazure** [comment]
The `try` statement and corresponding `catch` block only counts as one statement, so the rule is behaving correctly and you could remove the braces from the `if` block:

```js
/* global foo, cb */
if (foo)
  try {
    foo();
  } catch (err) {
    cb(err);
  }
```

If you find that unintuitive, you might prefer the ["multi-line" option](http://eslint.org/docs/rules/curly#multi-line).

**Comment by RyanZim** [comment]
> You might prefer the "multi-line" option.

The reason I'm not using multi-line is that I want to force
```
if (err) cb(err);
```
and disalow
```
if (err) { cb(err); }
```
Is there a way to do that?

**Comment by platinumazure** [comment]
Ack, you're right. Maybe try [multi-or-nest](http://eslint.org/docs/rules/curly#multi-or-nest)?

**Comment by RyanZim** [comment]
I don't like:
```
if (foo)
    foo++;
```
Should be:
```
if (foo) foo++;
```

**Comment by RyanZim** [comment]
OK, I ended up wrapping my conditional inside the try block instead of putting the try statement in the conditional.

```
/* global foo, cb */

try {
  if (foo) foo();
} catch (err) {
  cb(err);
}
```

Closing for now.
